src/coreclr/jit/morph.cpp

```diff
@@ -3179,7 +3181,9 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* call)
         }
 
         // TODO-ARGS: Review this, is it really necessary to treat them specially here?
-        if (call->gtArgs.IsNonStandard(this, call, &arg) && arg.AbiInfo.IsPassedInRegisters())
+        // Exception: Lower SwiftSelf struct arg to GT_LCL_FLD
+        if (call->gtArgs.IsNonStandard(this, call, &arg) && arg.AbiInfo.IsPassedInRegisters() &&
+            (arg.GetWellKnownArg() != WellKnownArg::SwiftSelf))
         {
```

Comment:
Why is this necessary? That doesn't seem right.

Reply:
Likely we should instead insert a new arg in the importer that represents "self" as a primitive instead of keeping this as a struct argument. It being a struct argument is just an artifact of the design and it wouldn't work to keep it as a struct on all ABIs.

Reply:
Let's just keep it as is for now. I'm refactoring stuff for the struct work which should make this change simple to make.

Reply:
> Likely we should instead insert a new arg in the importer that represents "self" as a primitive instead of keeping this as a struct argument.

Yeah, the fact that the argument was being kept as `TYP_STRUCT` was causing asserts in later phases.

> Let's just keep it as is for now. I'm refactoring stuff for the struct work which should make this change simple to make.

Sure thing.